[BUG] `request.response()` timeouts inside `context.route()`

**Context:**
- Playwright Version: `playwright@1.21.1`
- Operating System: Mac
- Node.js version: 14.18.1
- Browser: chromium

`npx envinfo --preset playwright --markdown`
## System:
 - OS: macOS 12.3.1
 - Memory: 359.18 MB / 16.00 GB
## Binaries:
 - Node: 14.18.1 - ~/.nvm/versions/node/v14.18.1/bin/node
 - Yarn: 1.22.18 - /usr/local/bin/yarn
 - npm: 6.14.15 - ~/.nvm/versions/node/v14.18.1/bin/npm
## Languages:
 - Bash: 3.2.57 - /bin/bash
## npmPackages:
 - playwright: ^1.21.1 => 1.21.1 

**Code Snippet**
```javascript
import { test } from '@playwright/test';

test('Intercepts', async ({ context, page }) => {
  await context.route('https://www.github.com/', async (route, request) => {
    console.log('Intercepted route: ', route.request().url());

    const response = await request.response();

    console.log('Route response: ', response);

    const body = await response.json();

    route.fulfill({
      status: 200,
      body: JSON.stringify(body),
    });
  });

  await page.goto('https://www.github.com/');
});
```

**Describe the bug**

When trying to `await request.response()` inside `context.route()`, the test timeouts.

Use case: I want intercept API request, get real server response, and change it.

*Actual Result*
```
> playwright test


Running 1 test using 1 worker

  _  [google-pixel] _ src/stub-response.spec.ts:3:1 _ Intercepts (30s)
Intercepted route:  https://www.github.com/
Route response:  null


  1) [google-pixel] _ src/stub-response.spec.ts:3:1 _ Intercepts ==================================

    Timeout of 30000ms exceeded.

    TypeError: Cannot read property 'json' of null

       9 |     console.log('Route response: ', response);
      10 |
    > 11 |     const body = await response.json();
         |                                 ^
      12 |
      13 |     route.fulfill({
      14 |       status: 200,

        at RouteHandler.handler (src/stub-response.spec.ts:11:33)

    Pending operations:
      - page.goto at src/stub-response.spec.ts:19:14
      - request.response at src/stub-response.ts:7:36


  Slow test file: [google-pixel] _ src/stub-response.spec.ts (30s)
  Consider splitting slow test files to speed up parallel execution

  1 failed
    [google-pixel] _ src/stub-response.spec.ts:3:1 _ Intercepts ===================================
npm ERR! Test failed.  See above for more details.
```

--- a/docs/src/api/class-route.md
+++ b/docs/src/api/class-route.md
@@ -3,6 +3,8 @@
 Whenever a network route is set up with [`method: Page.route`] or [`method: BrowserContext.route`], the `Route` object
 allows to handle the route.
 
+Learn more about [networking](../network.md).
+
 ## async method: Route.abort
 
 Aborts the route's request.

--- a/packages/playwright-core/types/types.d.ts
+++ b/packages/playwright-core/types/types.d.ts
@@ -14926,6 +14926,8 @@ export interface Response {
  * [page.route(url, handler[, options])](https://playwright.dev/docs/api/class-page#page-route) or
  * [browserContext.route(url, handler[, options])](https://playwright.dev/docs/api/class-browsercontext#browser-context-route),
  * the `Route` object allows to handle the route.
+ *
+ * Learn more about [networking](https://playwright.dev/docs/network).
  */
 export interface Route {
   /**

--- a/tests/config/experimental.d.ts
+++ b/tests/config/experimental.d.ts
@@ -14928,6 +14928,8 @@ export interface Response {
  * [page.route(url, handler[, options])](https://playwright.dev/docs/api/class-page#page-route) or
  * [browserContext.route(url, handler[, options])](https://playwright.dev/docs/api/class-browsercontext#browser-context-route),
  * the `Route` object allows to handle the route.
+ *
+ * Learn more about [networking](https://playwright.dev/docs/network).
  */
 export interface Route {
   /**

